pkg/engine/params_k8s.go

```diff
@@ -44,7 +46,7 @@ func assignKubernetesParameters(properties *api.Properties, parametersMap params
 				addValue(parametersMap, "kubernetesCcmImageSpec", kubernetesCcmSpec)
 			}
 
-			kubernetesHyperkubeSpec := hyperKubeImageBase + k8sComponents["hyperkube"]
+			kubernetesHyperkubeSpec := hyperkubeImageBase + hyperkubeComponent
```

Comment:
Please add unit test to cover this logic

Reply:
The PR notes explain that I need help with that. This change does not add a new execution path though.

Reply:
instead of needing a new `k8sComponents`  entry, you could also do:

```
if properties.IsAzureStackCloud() {
    kubernetesHyperkubeSpec := hyperKubeImageBase + k8sComponents["hyperkube"] + AzureStackSuffix
}
```

where `AzureStackSuffix` is a constant in https://github.com/Azure/aks-engine/blob/master/pkg/engine/const.go (and it can also be reused in armvariables.go)